From 8d3421dd744b0fff95a53c08cac4c33718bb98fc Mon Sep 17 00:00:00 2001
From: Benergy Meenan Ravuri <bravuri@codeaurora.org>
Date: Wed, 2 Dec 2015 12:29:07 +0530
Subject: [PATCH] IMS: Remove the methods with dummy implementation

Since the base implementation is added for the listener, the methods
with dummy implementation can be removed.

Change-Id: Iac9e75c53b362593c8cfd4e46ca0933566421aed
CRs-Fixed: 901766
---
 src/com/android/incallui/AnswerPresenter.java | 28 +++++----------------------
 1 file changed, 5 insertions(+), 23 deletions(-)

diff --git a/src/com/android/incallui/AnswerPresenter.java b/src/com/android/incallui/AnswerPresenter.java
index 2826205..020d075 100644
--- a/src/com/android/incallui/AnswerPresenter.java
+++ b/src/com/android/incallui/AnswerPresenter.java
@@ -32,6 +32,7 @@
 import com.cyanogen.lookup.phonenumber.provider.LookupProviderImpl;
 import org.codeaurora.ims.qtiims.IQtiImsInterface;
 import org.codeaurora.ims.qtiims.IQtiImsInterfaceListener;
+import org.codeaurora.ims.qtiims.QtiImsInterfaceListenerBaseImpl;
 import org.codeaurora.ims.qtiims.QtiImsInterfaceUtils;
 import org.codeaurora.ims.qtiims.QtiViceInfo;
 import org.codeaurora.QtiVideoCallConstants;
@@ -97,31 +98,12 @@ public void onServiceDisconnected(ComponentName className) {
         }
     };
 
-    /* IQtiImsInterfaceListener instance to handle call deflection response */
-    private IQtiImsInterfaceListener imsInterfaceListener = new IQtiImsInterfaceListener.Stub() {
-        public void onSetCallForwardUncondTimer(int status) {
-            /* Not implemented, dummy implementation to avoid compilation errors */
-        }
-
-        public void onGetCallForwardUncondTimer(int startHour, int endHour,
-                int startMinute, int endMinute, int reason, int status,
-                String number, int serviceClass) {
-            /* Not implemented, dummy implementation to avoid compilation errors */
-        }
-
-        public void onUTReqFailed(int errCode, String errString) {
-            /* Not implemented, dummy implementation to avoid compilation errors */
-        }
-
-        public void onGetPacketCount(int status, long packetCount) {
-            /* Not implemented, dummy implementation to avoid compilation errors */
-        }
-
-        public void onGetPacketErrorCount(int status, long packetErrorCount) {
-            /* Not implemented, dummy implementation to avoid compilation errors */
-        }
+    /* QtiImsInterfaceListenerBaseImpl instance to handle call deflection response */
+    private QtiImsInterfaceListenerBaseImpl imsInterfaceListener =
+            new QtiImsInterfaceListenerBaseImpl() {
 
         /* Handles call deflect response */
+        @Override
         public void receiveCallDeflectResponse(int result) {
             Log.w(this, "receiveCallDeflectResponse: " + result);
         }
